hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/scm/pipeline/TestPipelineManagerMXBean.java

```diff
@@ -66,29 +65,35 @@ public void init()
   public void testPipelineInfo() throws Exception {
     ObjectName bean = new ObjectName(
         "Hadoop:service=SCMPipelineManager,name=SCMPipelineManagerInfo");
+    Map<String, Integer> pipelineStateCount = cluster
+        .getStorageContainerManager().getPipelineManager().getPipelineInfo();
 
-    TabularData data = (TabularData) mbs.getAttribute(bean, "PipelineInfo");
-    Map<String, Integer> datanodeInfo = cluster.getStorageContainerManager()
-        .getPipelineManager().getPipelineInfo();
-    verifyEquals(data, datanodeInfo);
+    GenericTestUtils.waitFor(() -> {
+      try {
+        final TabularData data = (TabularData) mbs.getAttribute(
+            bean, "PipelineInfo");
+        for (Map.Entry<String, Integer> entry : pipelineStateCount.entrySet()) {
+          final Integer count = pipelineStateCount.get(entry.getKey());
```

Comment:
This still has the problem findbugs complained about previously:

```suggestion
        for (Map.Entry<String, Integer> entry : pipelineStateCount.entrySet()) {
          final Integer count = entry.getValue();
```

(For some reason findbugs no longer complains about it.)

Reply:
Pls don't review yet, I am aware of these findbugs coming. On my local machine, findbugs plugin is not working properly, I'll fix these. Let me convert the PR to draft.